// react-responsive-pagination/lib.cjs/hooks/usePaginationItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePaginationItems = void 0;
var react_1 = require("react");
var compositions_1 = require("../compositions");
var util_1 = require("../helpers/util");
var paginationItem_1 = require("../paginationItem");
var useWidestComposition_1 = require("./useWidestComposition");
function usePaginationItems(inputCurrent, inputTotal, maxWidth, options) {
    var narrowToWideCompositionsProvider = function () {
        var _a, _b, _c;
        return (0, compositions_1.narrowToWideCompositions)({
            current: (_a = (0, util_1.sanatizeInteger)(inputCurrent)) !== null && _a !== void 0 ? _a : 0,
            total: (_b = (0, util_1.sanatizeInteger)(inputTotal)) !== null && _b !== void 0 ? _b : 0,
            narrowStrategies: sanatizeNarrowStrategies(options === null || options === void 0 ? void 0 : options.narrowStrategy),
            renderNav: (_c = (0, util_1.sanatizeBoolean)(options === null || options === void 0 ? void 0 : options.renderNav)) !== null && _c !== void 0 ? _c : true,
        });
    };
    var _a = (0, useWidestComposition_1.useWidestComposition)(narrowToWideCompositionsProvider, maxWidth), compositionItems = _a.items, ref = _a.ref, clearCache = _a.clearCache;
    (0, react_1.useEffect)(function () {
        return function () { return clearCache(); };
    }, [clearCache, options === null || options === void 0 ? void 0 : options.previousLabel, options === null || options === void 0 ? void 0 : options.nextLabel]);
    var items = (0, paginationItem_1.compositionToPaginationItems)(compositionItems, options);
    return { items: items, ref: ref, clearCache: clearCache };
}
exports.usePaginationItems = usePaginationItems;
function sanatizeNarrowStrategies(inputNarrowStrategies) {
    return (Array.isArray(inputNarrowStrategies)
        ? inputNarrowStrategies
        : [inputNarrowStrategies]).filter(function (strategy) { return strategy === 'dropEllipsis' || strategy === 'dropNav'; });
}
